[CPU] Enchant Edge "to string" logic

EgorDuplensky opened the pull request:
by replacing "name()" with ostream operator.
Also include "status" to the printed information.

EgorDuplensky commented:
@dmitry-gorokhov Could you plese take a look

dmitry-gorokhov reviewed (approved):
@EgorDuplensky I am ok with the changes. Just fro my understanding: what is rationale behind it?

EgorDuplensky commented:
> @EgorDuplensky I am ok with the changes. Just fro my understanding: what is rationale behind it?

The base idea is just to include Status into the edge toString logic.
Rest of the changes are just consequences of that idea. For example, we cannot really include Status into edge "name()" method, because this method is actually being used as a key in weights cache storage. So, to avoid any confusion, I have renamed `name()` method into the `hash()` and implemented operator overloading as it is usually done
